index.js

```diff
@@ -168,10 +169,15 @@ function load(path, options, iterator) {
 }
 
 
-function generate(path, options, md) {
+function generate(path, options, env, md) {
```

Comment:
1. I suspect that will break existing API.
2. If you passed `options` OR `env` (only one), that will fail to detect param type

Why not add `env` as last param?